internal/provider/gcp_labels_data_source.go

```diff
@@ -0,0 +1,191 @@
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "GCP Labels data source",
+
+		Attributes: map[string]schema.Attribute{
+			"tags": schema.MapAttribute{
+				MarkdownDescription: "Map of tags.",
+				Computed:            true,
+				ElementType:         types.StringType,
+			},
+			"tags_as_list": schema.ListAttribute{
+				MarkdownDescription: "List of tags in {Key='key', Value='value'} format.",
+				Computed:            true,
+				ElementType:         types.MapType{ElemType: types.StringType},
+			},
+			"tags_key_case": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "The case to use for the keys of tags created by the provider. Valid values are: none, camel, lower, snake, title, upper.",
+				Validators: []validator.String{
+					stringvalidator.OneOf(ValidCases...),
+				},
+			},
+			"tags_value_case": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "The case to use for the values of tags created by the provider. Valid values are: none, camel, lower, snake, title, upper.",
+				Validators: []validator.String{
+					stringvalidator.OneOf(ValidCases...),
+				},
+			},
+			"values": schema.MapAttribute{
+				MarkdownDescription: "Map of values to override or add to the context when creating the label.",
+				Optional:            true,
+				ElementType:         types.StringType,
+			},
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Tags identifier",
+				Computed:            true,
+			},
+			"replacement_map": schema.MapAttribute{
+				MarkdownDescription: "Map of strings to replace in the tag, applies to both key and value. The key is the string to replace, and the value is the string to replace it with.",
+				Optional:            true,
+				ElementType:         types.StringType,
+			},
+		},
+	}
+}
+
+//nolint:gocritic
+func (d *GcpLabelsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var config GcpLabelsDataSourceModel
+
+	// Read Terraform configuration data into the model.
+	resp.Diagnostics.Append(req.Config.Get(ctx, &config)...)
+
+	localValues := d.getLocalValues(ctx, &config.TagsDataSourceModel, resp)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	localTagsKeyCase := d.getLocalTagsKeyCase(&config.TagsDataSourceModel, resp)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	localTagsValueCase := d.getLocalTagsValueCase(&config.TagsDataSourceModel, resp)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	d.setTags(ctx, &config, resp, localValues, localTagsKeyCase, localTagsValueCase)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	d.setTagsList(ctx, &config, resp, localValues, localTagsKeyCase, localTagsValueCase)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, "created GCP labels data source")
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &config)...)
+}
+
+//nolint:revive
```

Comment:
What is being turned off here? Is there no other way to deal with the lint error?